src/System.Security.Cryptography.Csp/tests/RSACryptoServiceProviderBackCompat.cs

```diff
@@ -116,6 +122,63 @@ public static void SignHash_PublicKey()
             }
         }
 
+        [Fact]
+        public static void OldVerifyHash()
+        {
+            byte[] dataHash, signature;
+
+            using (HashAlgorithm hash = SHA256.Create())
+            {
+                dataHash = hash.ComputeHash(TestData.HelloBytes);
+            }
+
+            // Use the new RSA SignHash method to generate the signature
+            using (RSA rsa = RSAFactory.Create())
+            {
+                rsa.ImportParameters(TestData.RSA2048Params);
+                signature = rsa.SignHash(dataHash, new HashAlgorithmName("SHA256"), RSASignaturePadding.Pkcs1);
+            }
+
+            // Verify using the old RSACryptoServiceProvider VerifyHash
+            using (var rsa = new RSACryptoServiceProvider())
+            {
+                rsa.ImportParameters(s_publicOnly);
+                Assert.True(rsa.VerifyHash(dataHash, "SHA256", signature));
+            }
+        }
+
+        [Fact]
+        public static void OldSignHash()
+        {
+            byte[] dataHash, signature;
+
+            using (HashAlgorithm hash = SHA256.Create())
+            {
+                dataHash = hash.ComputeHash(TestData.HelloBytes);
+            }
+
+            // Use the old RSACryptoServiceProvider SignHash method to generate the signature
+            using (var rsa = new RSACryptoServiceProvider())
+            {
+                rsa.ImportParameters(TestData.RSA2048Params);
+                signature = rsa.SignHash(dataHash, "SHA256");
+            }
+
+            // Verify using the old RSACryptoServiceProvider VerifyHash
+            using (var rsa = new RSACryptoServiceProvider())
+            {
+                rsa.ImportParameters(TestData.RSA2048Params);
+                Assert.True(rsa.VerifyHash(dataHash, "SHA256", signature));
+            }
```

Comment:
Was this a copy/paste error? Or were you explicitly doing two verify calls for one sign?  If you're claiming utility in ensuring old-sign works with old-verify, maybe these two methods can be squished into one theory:

```C#
[Theory]
[InlineData(true, true)]
[InlineData(true, false)]
[InlineData(false, true)]
// (false, false) is not required, that would be equivalent to the RSA AlgorithmImplementation suite.
public static void VerifyLegacySignVerifyHash(bool useLegacySign, bool useLegacyVerify)
{
    byte[] dataHash, signature;

    using (HashAlgorithm hash = SHA256.Create())
    {
        dataHash = hash.ComputeHash(TestData.HelloBytes);
    }

    using (var rsa = new RSACryptoServiceProvider())
    {
        rsa.ImportParameters(TestData.RSA2048Params);

        signature = useLegacySign ?
            rsa.SignHash(dataHash, "SHA256") :
            rsa.SignHash(dataHash, HashAlgorithmName.SHA256, RSASignaturePadding.Pkcs1);
    }

    bool verified;

    using (var rsa = new RSACryptoServiceProvider())
    {
        rsa.ImportParameters(
            new RSAParameters
            {
                Modulus = TestData.RSA2048Params.Modulus,
                Exponent = TestData.RSA2048Params.Exponent,
            });

        verified = useLegacyVerify ?
            rsa.VerifyHash(dataHash, "SHA256", signature) :
            rsa.VerifyHash(dataHash, signature, HashAlgorithmName.SHA256, RSASignaturePadding.Pkcs1);
    }

    Assert.True(verified);
}

Heck, maybe it should be that theory method anyways, whether or not (true, true) makes it into the list.